core/src/main/java/org/apache/carbondata/core/cache/CacheProvider.java

```diff
@@ -84,14 +84,13 @@ public static CacheProvider getInstance() {
     //check if cache is null for given cache type, if null create one
     if (!dictionaryCacheAlreadyExists(cacheType)) {
       synchronized (lock) {
-        if (!dictionaryCacheAlreadyExists(cacheType)) {
-          if (null == cacheTypeToLRUCacheMap.get(cacheType)) {
-            createLRULevelCacheInstance(cacheType);
-          }
-          createDictionaryCacheForGivenType(cacheType, carbonStorePath);
+        if (null == cacheTypeToLRUCacheMap.get(cacheType)) {
+          createLRULevelCacheInstance(cacheType);
         }
+        createDictionaryCacheForGivenType(cacheType, carbonStorePath);
```

Comment:
@PallaviSingh1992 ....dictionaryCacheAlreadyExists will check if for a given cache type (Forward/Reverse), the object has already been created. Now this check is before and inside synchronized block to handle concurrent access (double locking mechanism). If this check is removed inside the synchronized block then it can happen that in concurrent scenarios, multiple threads end up in creating multiple objects for the same cache type and the object in the map keeps getting overridden. 
Therefore I think that this double check is required. Please correct me if I am wrong.